src/lib/cache/subCache.cpp

```diff
@@ -0,0 +1,1105 @@
+  cSubP->count                 = (notificationDone == true)? 1 : 0;
+
+  LM_T(LmtSubCache, ("inserting a new sub in cache (%s). lastNotifictionTime: %lu", cSubP->subscriptionId, cSubP->lastNotificationTime));
+
+
+  //
+  // 2. Then the values that have functions/methods for filling/parsing
+  //
+  cSubP->throttling = scrP->throttling.parse();
+  cSubP->notifyConditionVector.fill(scrP->notifyConditionVector);
+
+
+  //
+  // 3. Convert all EntityIds to EntityInfo
+  //
+  for (unsigned int ix = 0; ix < scrP->entityIdVector.vec.size(); ++ix)
+  {
+    EntityId* eIdP;
+
+    eIdP = scrP->entityIdVector.vec[ix];
+
+    EntityInfo* eP = new EntityInfo(eIdP->id, eIdP->type, eIdP->isPattern);
+    
+    cSubP->entityIdInfos.push_back(eP);
+  }
+
+
+  //
+  // 4. Insert all attributes 
+  //
+  for (unsigned int ix = 0; ix < scrP->attributeList.attributeV.size(); ++ix)
+  {
+    cSubP->attributes.push_back(scrP->attributeList.attributeV[ix]);
+  }
+
+
+  //
+  // 5. Now, insert the subscription in the cache
+  //
+  LM_T(LmtSubCache, ("Inserting NEW sub '%s', lastNotificationTime: %lu", cSubP->subscriptionId, cSubP->lastNotificationTime));
+  subCacheItemInsert(cSubP);
+}
+
+
+
+/* ****************************************************************************
+*
+* subCacheStatisticsGet - 
+*/
+void subCacheStatisticsGet(int* refreshes, int* inserts, int* removes, int* updates, int* items, char* list, int listSize)
+{
+  *refreshes = subCache.noOfRefreshes;
+  *inserts   = subCache.noOfInserts;
+  *removes   = subCache.noOfRemoves;
+  *updates   = subCache.noOfUpdates;
+  *items     = subCacheItems();
+
+  CachedSubscription* cSubP = subCache.head;
+
+  //
+  // NOTE
+  //   If the listBuffer is big enough to hold the entire list of cached subscriptions,
+  //   subCacheStatisticsGet returns that list for the response of "GET /cache/statistics".
+  //   Minimum size of the list is set to 128 bytes and the needed size is detected later.
+  //   If the list-buffer is not big enough to hold the entire list, a warning text is showed instead.
+  //   
+  //
+  *list = 0;
+  if (listSize > 128)
+  {
+    while (cSubP != NULL)
+    {
+      char          msg[256];
+      unsigned int  bytesLeft = listSize - strlen(list);
+
+#if 0
+      //
+      // FIXME P5: To be removed once sub-update is OK in QA
+      //
+      int now = getCurrentTime();
+      snprintf(msg, sizeof(msg), "%s|N:%lu|E:%lu|T:%lu|DUR:%lu",
+               cSubP->subscriptionId,
+               cSubP->lastNotificationTime,
+               cSubP->expirationTime,
+               cSubP->throttling,
+               cSubP->expirationTime - now);
+#else
+        snprintf(msg, sizeof(msg), "%s", cSubP->subscriptionId);
+#endif
+      //
+      // If "msg" and ", " has no room in the "list", then no list is shown.
+      // (the '+ 2' if for the string ", ")
+      //
+      if (strlen(msg) + 2 > bytesLeft)
+      {
+        snprintf(list, listSize, "too many subscriptions");
```

Comment:
Needs to mention this in CNR? I would depend if the item list functionality in cache statistics was already in 0.25.0...